client/src/extension.ts

```diff
@@ -96,9 +96,7 @@ export function activate(context: ExtensionContext) {
       // We'll send the initial configuration in here, but this might be
       // problematic because every consumer of the LS will need to mimic this.
       // We'll leave it like this for now, but might be worth revisiting later on.
-      initializationOptions: {
-        extensionConfiguration: workspace.getConfiguration("rescript.settings"),
-      },
+      initializationOptions: workspace.getConfiguration("rescript.settings"),

```

Comment:
While I appreciate wanting to keep thing simple, it's a choice whether we want to break user's configuration now or later. What if we decide we want more things sent to the server from the client at a later stage? Could be as simple as meta data around _what_ client it is, or whatever. If we make this change, we'd need to migrate back to the current solution again to support that, and that'd break all clients. I think the current approach is more future proof, even if it means sending an extra prop.